### PR TITLE
resource_manager: save tokens in persistent storage

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #5851 

### What is changed and how does it work?
save GroupTokenBucketState every 1 minite

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

Code changes
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 1]
```diff
@@ -145,17 +145,23 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
 	switch group.GetMode() {
 	case rmpb.GroupMode_RUMode:
 		if settings := group.GetRUSettings(); settings != nil {
-			ruSettings = &RequestUnitSettings{
-				RU: NewGroupTokenBucket(settings.GetRU()),
+			ruSettings = &RequestUnitSettings{}
+			if settings.GetRU() != nil {
```
Comment (BornChanger) [inline pkg/mcs/resource_manager/server/resource_group.go:149]: what will happen if settings.GetRU() is nil?

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 2]
```diff
@@ -164,10 +170,17 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
 }
 
 // RequestRU requests the RU of the resource group.
-func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64) *rmpb.GrantedRUTokenBucket {
+func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64, storage endpoint.ResourceGroupStorage) *rmpb.GrantedRUTokenBucket {
 	rg.Lock()
 	defer rg.Unlock()
-	if rg.RUSettings == nil {
+	grant := rg.requestRU(now, neededTokens, targetPeriodMs)
```
Comment (BornChanger) [inline pkg/mcs/resource_manager/server/resource_group.go:176]: How much overhead will it pay here? And how often is it called and do we need to call it so often?

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 3]
```diff
@@ -164,10 +170,17 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
...
+	rg.persistStates(storage)
```
Comment (BornChanger) [inline pkg/mcs/resource_manager/server/resource_group.go:177]: How much overhead will it pay here? And how often is it called and do we need to call it so often?

File: pkg/mcs/resource_manager/server/resource_group.go (hunk 1 above)
Comment (CabinfeverB) [reply pkg/mcs/resource_manager/server/resource_group.go:149]: `RUSettings.RU` will be an empty GroupTokenBucket which has nil pointer of `Settings`.  If `Settings` is a nil pointer, GroupTokenBucket can not handle RequestRU

**Review by nolouch — APPROVED** [review]
lgtm. 
maybe we can improve it if the group no updates in the future.

**Review by BornChanger — APPROVED** [review]

**Comment by nolouch** [comment]
ptal @JmPotato

**Review by JmPotato — APPROVED** [review]

**Pull request merged**